content/100-getting-started/02-setup-prisma/100-start-from-scratch/110-relational-databases/100-connect-your-database.mdx

```diff
@@ -111,6 +111,59 @@ DATABASE_URL="mysql://root:randompassword@localhost:3306/mydb"
 
 </SwitchTech>
 
+<SwitchTech technologies={['*', 'planetscale']}>
+
+```prisma file=prisma/schema.prisma
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+```
+
+Note that the default schema created by `prisma init` uses PostgreSQL as the `provider`. For PlanetScale, you need to use the `mysql` provider instead:
+
+```prisma file=prisma/schema.prisma highlight=2;edit
+datasource db {
+  provider = "mysql"
+  url      = env("DATABASE_URL")
+}
+```
+
+In this case, the `url` is [set via an environment variable](/concepts/components/prisma-schema#accessing-environment-variables-from-the-schema) which is defined in `.env`:
+
+```bash file=.env
+DATABASE_URL="mysql://janedoe:mypassword@planetscale-hostname/mydb?sslaccept=strict"
+```
+
+You now need to adjust the connection URL to point to your own database.
+
+The format of the connection URL for your database typically depends on the database you use. PlanetScale uses the MySQL connection URL format, which has the following structure (the parts spelled all-uppercased are _placeholders_ for your specific connection details):
+
+```no-lines
+mysql://USER:PASSWORD@HOST:PORT/DATABASE
+```
+
+Here's a short explanation of each component:
+
+- `USER`: The name of your database user
+- `PASSWORD`: The password for your database user
+- `PORT`: The port where your database server is running (typically `3306` for MySQL)
+- `DATABASE`: The name of the [database](https://dev.mysql.com/doc/refman/8.0/en/creating-database.html)
+
+For a database hosted with PlanetScale, the [connection URL](/reference/database-reference/connection-urls) might look similar to this:
+
+```bash file=.env
+DATABASE_URL="mysql://johndoe:XXX@123456789abc.eu-west-3.psdb.cloud:3306/mydb"
+```
+
+When connecting to PlanetScale locally, your connection URL typically looks similar to this:
+
+```bash file=.env
+DATABASE_URL="mysql://johndoe:XXX@localhost:3306/mydb"
```

Comment:
I think the username and password when using `pscale` (which is the way to connect to PlanetScale via a local proxy) was define - we should double check that.

Reply:
Can I check what this means - does it mean that the username and password are autogenerated for you?

Reply:
With connection strings you can get from the PlanetScale UI, this is not a super important workflow any more.

But if you do it, it looks like this:
```
λ pscale connect netlify-starter main
Tried address 127.0.0.1:3306, but it's already in use. Picking up a random port ...
Secure connection to database migration-table and branch main is established!.

Local address to connect your application: 127.0.0.1:49395 (press ctrl-c to quit)
```

So in my case this would lead to a connection string similar to this:
```
mysql://root@127.0.0.1:49395/netlify-starter
```

(I can not even find proper documentation about this any more for the PlanetScale CLI)

Reply:
I'm still confused by this, sorry - I don't think I understand what 'was define' means in your first comment. When I tested using the CLI I got a connection string in the format

`mysql://username:password@localhost:3306/prisma-docs-test`

with `username` and `password` autogenerated

Reply:
That was a typo, I meant "was defined" in that you could not actually choose the username and password when using `pscale connect`.

(You do get a username and password if you create a connection string via the CLI on PlanetScale - but that is then not with `localhost`)

Can you post the command you were using and the output you got? Maybe I am just missing something.

Reply:
What I did was:

```
pscale auth login
pscale connect prisma-docs-test develop
```
Then I added the `DATABASE_URL` to my `.env` file:

```
mysql://username:password@localhost:3306/prisma-docs-test
```

with a username and password that I'd got from the UI. So kind of a weird hybrid between using the UI and the CLI - I was roughly following [this blog post](https://docs.planetscale.com/tutorials/automatic-prisma-migrations) but obviously diverged there.

I'm now confused as to why this works, as when I look in settings the username and password I used are for the `main` branch... but I'm definitely connected to `develop`, and my schema changes have been pushed there.